steering_controllers/steering_controllers/src/steering_controllers.cpp

```diff
@@ -0,0 +1,515 @@
+      e.what());
+    return controller_interface::CallbackReturn::ERROR;
+  }
+
+  // TODO(anyone): Reserve memory in state publisher depending on the message type
+  rt_odom_state_publisher_->lock();
+  rt_odom_state_publisher_->msg_.header.stamp = get_node()->now();
+  rt_odom_state_publisher_->msg_.header.frame_id = params_.odom_frame_id;
+  rt_odom_state_publisher_->msg_.child_frame_id = params_.base_frame_id;
+  rt_odom_state_publisher_->msg_.pose.pose.position.z = 0;
+
+  auto & covariance = rt_odom_state_publisher_->msg_.twist.covariance;
+  constexpr size_t NUM_DIMENSIONS = 6;
+  for (size_t index = 0; index < 6; ++index)
+  {
+    // 0, 7, 14, 21, 28, 35
+    const size_t diagonal_index = NUM_DIMENSIONS * index + index;
+    covariance[diagonal_index] = params_.pose_covariance_diagonal[index];
+    covariance[diagonal_index] = params_.twist_covariance_diagonal[index];
+  }
+  rt_odom_state_publisher_->unlock();
+
+  try
+  {
+    // Tf State publisher
+    tf_odom_s_publisher_ = get_node()->create_publisher<ControllerStateMsgTf>(
+      "~/tf_odometry", rclcpp::SystemDefaultsQoS());
+    rt_tf_odom_state_publisher_ =
+      std::make_unique<ControllerStatePublisherTf>(tf_odom_s_publisher_);
+  }
+  catch (const std::exception & e)
+  {
+    fprintf(
+      stderr, "Exception thrown during publisher creation at configure stage with message : %s \n",
+      e.what());
+    return controller_interface::CallbackReturn::ERROR;
+  }
+
+  rt_tf_odom_state_publisher_->lock();
+  rt_tf_odom_state_publisher_->msg_.transforms.resize(1);
+  rt_tf_odom_state_publisher_->msg_.transforms[0].header.stamp = get_node()->now();
+  rt_tf_odom_state_publisher_->msg_.transforms[0].header.frame_id = params_.odom_frame_id;
+  rt_tf_odom_state_publisher_->msg_.transforms[0].child_frame_id = params_.base_frame_id;
+  rt_tf_odom_state_publisher_->msg_.transforms[0].transform.translation.z = 0.0;
+  rt_tf_odom_state_publisher_->unlock();
+
+  try
+  {
+    // State publisher
+    controller_s_publisher_ = get_node()->create_publisher<AckermanControllerState>(
+      "~/controller_state", rclcpp::SystemDefaultsQoS());
+    controller_state_publisher_ =
+      std::make_unique<ControllerStatePublisher>(controller_s_publisher_);
+  }
+  catch (const std::exception & e)
+  {
+    fprintf(
+      stderr, "Exception thrown during publisher creation at configure stage with message : %s \n",
+      e.what());
+    return controller_interface::CallbackReturn::ERROR;
+  }
+
+  // TODO(anyone): Reserve memory in state publisher depending on the message type
+  controller_state_publisher_->lock();
+  controller_state_publisher_->msg_.header.stamp = get_node()->now();
+  controller_state_publisher_->msg_.header.frame_id = params_.odom_frame_id;
+  controller_state_publisher_->msg_.odom.pose.pose.position.z = 0;
+  controller_state_publisher_->msg_.odom.child_frame_id = params_.base_frame_id;
+  controller_state_publisher_->unlock();
```

Comment:
Is this actually needed?

Reply:
Not sure. We publish a lot of states/statuses and have publishers with a large overlap at the moment. Not sure do we need `odom_s_publisher_`, `tf_odom_s_publisher_` and ```
 using AckermanControllerState = control_msgs::msg::SteeringControllerStatus;
  AckermanControllerState published_state_;
```?